subsys/testsuite/ztest/include/zephyr/ztest_test_new.h

```diff
@@ -53,6 +53,10 @@ struct ztest_suite_stats {
 /**
  * Setup function to run before running this suite
  *
+ * Note: Never zassert in the setup. As this will kill the main test thread causing the test process
+ * to hang. Instead, use zassume (for behavior tested elsewhere, see the zassume function doc) or
+ * add the zassert as its own test or as part of the before/after functions.
```

Comment:
This is behavior is being changed with https://github.com/zephyrproject-rtos/zephyr/pull/48768 to allow asserting anywhere. Please update comment to reflect this.

Reply:
Updated